support-macros/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+#![recursion_limit = "128"]
+extern crate proc_macro;
+extern crate proc_macro2;
+
+use proc_macro::TokenStream;
+
+use quote::quote;
+use syn;
+use syn::Error;
+use syn::ItemFn;
+
+#[proc_macro_attribute]
+pub fn libtock_main(_: TokenStream, input: TokenStream) -> TokenStream {
+    generate_main_wrapped(input.into()).into()
+}
+
+fn generate_main_wrapped(input: proc_macro2::TokenStream) -> proc_macro2::TokenStream {
+    try_generate_main_wrapped(input).unwrap_or_else(|err| err.to_compile_error().into())
+}
+
+fn try_generate_main_wrapped(
+    input: proc_macro2::TokenStream,
+) -> Result<proc_macro2::TokenStream, Error> {
+    let ast = syn::parse2::<ItemFn>(input)?;
+    let block = ast.block;
+    Ok(quote!(
+        fn main() -> ::libtock::result::TockResult<()> {
+            unsafe {::core::executor::block_on( async #block ) }
```

Comment:
The code should panic if `main` does not exit successfully. Calling `.ok().unwrap()` would be okay for me.

Reply:
I think the error message for that would look like `unwrap() called on None value at <???>`, and I'd prefer a better message than that. We should probably revive `Termination` to print out a useful error message if the return value is `Err(...)`.

Reply:
Now that I think about it: Panicking isn't a good idea. It will create a complex object containing information about where the code crashed. But printing a fixed message, on the other hand, also has some overhead.